File: python/tests/test_data.py
import pytest

from rts_gmlc_case import data


def test_source_dir_exists_and_has_tables():
    src = data.rts_source_dir()
    for name in data.REQUIRED_SOURCE_TABLES:
        assert (src / data.SOURCE_DATA / name).is_file()
    assert (src / data.TIMESERIES_DATA).is_dir()


def test_partial_cache_raises_with_missing_names(tmp_path, monkeypatch):
    monkeypatch.setattr(data, "REPO_ROOT", tmp_path)
    source_data = tmp_path / "data" / "RTS-GMLC" / "RTS_Data" / data.SOURCE_DATA
    source_data.mkdir(parents=True)
    (source_data / "bus.csv").write_text("")

    with pytest.raises(RuntimeError) as exc_info:
        data.rts_source_dir()

    message = str(exc_info.value)
    for name in ("branch.csv", "gen.csv", "dc_branch.csv", "reserves.csv", "storage.csv", "timeseries_pointers.csv"):
        assert name in message
    assert data.TIMESERIES_DATA in message

File: julia/RTSGMLCCase/test/test_data.jl
@testset "source data" begin
    src = RTSGMLCCase.rts_source_dir()
    for name in RTSGMLCCase.REQUIRED_SOURCE_TABLES
        @test isfile(joinpath(src, RTSGMLCCase.SOURCE_DATA, name))
    end
    @test isdir(joinpath(src, RTSGMLCCase.TIMESERIES_DATA))
end

@testset "partial cache raises with missing names" begin
    mktempdir() do repo_root
        source_data = joinpath(repo_root, "data", "RTS-GMLC", "RTS_Data", RTSGMLCCase.SOURCE_DATA)
        mkpath(source_data)
        write(joinpath(source_data, "bus.csv"), "")

        err = nothing
        try
            RTSGMLCCase.rts_source_dir(repo_root)
        catch caught
            err = caught
        end
        @test err !== nothing
        message = sprint(showerror, err)
        for name in
            ("branch.csv", "gen.csv", "dc_branch.csv", "reserves.csv", "storage.csv", "timeseries_pointers.csv")
            @test occursin(name, message)
        end
        @test occursin(RTSGMLCCase.TIMESERIES_DATA, message)
    end
end
